### DISCUSSION TITLE
Does REL support association between Table ?

### DISCUSSION THREAD

**Discussion by wavemoroc001** [go-rel/rel#301] [Ideas]
like many to many, many to one

**Comment by Fs02** [comment]
REL have some support for loading and saving association
you can find it in the documentation: https://go-rel.github.io/association/

note: many to many / join through table is not currently supported, but workaround by defining join table as struct should be do able

**Reply by Rudis1261** [reply]
@Fs02 First off, love the project and I have really been enjoying using it! 

If I can please ask based on your last comment, is there an example somewhere showing how to do this? 

For my use-case, I have a couple of many-to-many relationships, but I can only get it to load one level deep.

**Reply by Fs02** [reply]
something like this:

```go
type User struct {
  ID int
  UserRoles []UserRole
}

type Role struct {
  ID int
}

type UserRole struct {
  UserID int
  User User
  RoleID int
  Role Role
}

// preload
repo.Preload(ctx, &user, "user_roles") // preload the join table first
repo.Preload(ctx, &user, "user_roles.role")

// now role can be accessed like this
user.user_role[0].role
```
